File: underscore.rb
def to_underscore(string)
  char_array = string.to_s.split('')
  for x in 1...char_array.length
    char_array[x] = '_' + if char_array[x] == char_array[x].upcase
  end
  p char_array
end


# returns test_controller
to_underscore('TestController')

# returns movies_and_books
to_underscore('MoviesAndBooks')

# returns app7_test
to_underscore('App7Test')

# returns "1"
to_underscore(1)

File: js/underscore.js
function toUnderscore(string) {
  var charArray = string.toString().split('');
  charArray[0] = charArray[0].toLowerCase();
  for (i = 1; i < charArray.length; i++) {
    if (charArray[i] === charArray[i].toUpperCase() &&
    charArray[i] !== charArray[i].toLowerCase()) {
      charArray.splice(i, 1, '_', charArray[i].toLowerCase());
      i++;
    };
  };
  return charArray.join('')
}
//  returns test_controller
toUnderscore('TestController');

// returns movies_and_books
toUnderscore('MoviesAndBooks');

// returns app7_test
toUnderscore('App7Test');

// returns "1"
toUnderscore(1);
